### ISSUE TITLE
[Accessibility] [Color Contrast] The Link is hard to see when the dataGridViewlinkcell is clicking and after clicking when the dataGridView row is highlighted/selected situation

### ISSUE THREAD

**Issue by VinoWang01** [dotnet/winforms#6116]
* .NET Core Version: .NET 7.0.100-alpha.1.21552.6

* Have you experienced this same bug with .NET Framework?: Yes

**Problem description:**
When user select/highlight the full row of DataGridView, then click the link in the link cell. The link is hard to see when clicking and after clicking it.
Clicking:
![Clicking](https://user-images.githubusercontent.com/90816462/140027888-a2af72b4-603d-44b5-9205-2bdd76b66326.png)

After Clicking:
![AfterClick](https://user-images.githubusercontent.com/90816462/140027896-20de0e31-53c9-43fa-8a44-6e6fdd56be2f.png)

**Expected behavior:**
The link should be easy to see.

**Minimal repro:**
1.	Create a Winforms .Net Core application with a DataGridView control. 
2.	Click Edit Columns... in DataGridView's smart tag. 
3.	Click Add button in the Edit Columns dialog. 
4.	Set Type property value to DataGridViewLinkColumn in Add Column dialog and click Close button. 
5.	Click the ellopsis… button of the DefaultCellStyle property of the Column1 in the Edit Columns dialog. 
6.	Set the NullValue property value to a website(https://www.baidu.com) in the CellStyle builder dialog and Click OK button. 
7.	Build and run it, click the link. 
8.	Launch the Accessibility insights tool to observe the color contrast of the link. 
Or 
1.	Extract sample project and open it in VS. [WindowsFormsApp3_7.0.zip](https://github.com/dotnet/winforms/files/7466151/WindowsFormsApp3_7.0.zip)
2.	Build and run the project, click the link. 
3.	Launch the Accessibility insights tool to observe the color contrast of the link.

**Comment by Danil-Andrianov** [comment]
@merriemcgaw Hi, Merrie! I have two options for this issue's resolving. Please, see the gifs below.
![ver1](https://user-images.githubusercontent.com/74228865/140065492-ea429f5d-1699-485f-a18b-7d087e3ce1b9.gif)
![ver2](https://user-images.githubusercontent.com/74228865/140065506-96b48a5b-26a1-4247-b71f-52085fbd803e.gif)
Maybe you could suggest better colors?

**Comment by vladimir-krestov** [comment]
A link in a DGV is a regular text, so its color should have contrast 4.5:1 (AI requirements)
For the current blue selection color only white color has correct contrast (4.508:1), other colors are not acceptable.
Changing of the selection color to lighter looks unshapely.
![image](https://user-images.githubusercontent.com/49272759/140927277-cb6c7e52-b9bf-4124-ba7c-ed69e4e37184.png)

So that is a difficult question. @merriemcgaw, could you please ask accessibility experts, which colors should we use?

**Comment by merriemcgaw** [comment]
I think white should be sufficient, was there a blocker in just using that @SergeySmirnov-Akvelon @vladimir-krestov ?

**Comment by Danil-Andrianov** [comment]
@merriemcgaw 
Hi, Merrie! Yes, there is blocker. When cell is selected, there are two different color states for link: VisitedLinkColor and ActiveLinkColor. One of them we could make white, but there is no suitable color for another state.
So I decided to change the color of the selected cell and made it darker. Please, see the result below.
![image](https://user-images.githubusercontent.com/74228865/141081609-1555f917-6c10-4217-9af1-19b29e756e60.png)
What do you think about it?

**Comment by merriemcgaw** [comment]
I don't think I like that dark selection - it doesn't match up with the selected cell color anywhere else. Was there a yellow we could use for Visited Link?

**Comment by Danil-Andrianov** [comment]
@merriemcgaw In case of yellow we'll lose the required contrast. Do you think 4:1 will be good enough?

**Comment by merriemcgaw** [comment]
I think we would have a technical violation, but I don't know how serious it is. Could we find some other way to indicate that the link has been followed rather than a new color? Maybe bold or italicize links that have been visited? That way we could keep the blue the way it is and the white will still follow contrast guidelines while making the links visibly different. It's not ideal, but it might be better than not having enough contrast. @Tanya-Solyanik what are your thoughts here?

**Comment by Philip-Wang01** [comment]
Verified this issue on .NET SDK 7.0.100-alpha.1.22060.5, this issue was fixed, the Link is easy to see when the dataGridViewlinkcell is clicking and after clicking when the dataGridView row is highlighted/selected situation.
![123456789](https://user-images.githubusercontent.com/86937911/148892412-9eba739e-23a3-429a-8722-587108618a85.gif)
